mne_bids/read.py

```diff
@@ -71,6 +72,92 @@ def _read_raw(raw_fname, electrode=None, hsp=None, hpi=None, config=None,
     return raw
 
 
+def _handle_events_reading(events_fname, raw):
+    """Read associated events.tsv and populate raw.
+
+    Handle onset, duration, and description of each event.
+    """
+    events_dict = _from_tsv(events_fname)
+
+    # Get the descriptions of the events
+    if 'trial_type' in events_dict:
+        # Drop events unrelated to a trial type
+        events_dict = _drop(events_dict, 'n/a', 'trial_type')
+        descriptions = np.asarray(events_dict['trial_type'], dtype=str)
+
+    # If we don't have a proper description of the events, perhaps we have
+    # at least an event value?
+    elif 'value' in events_dict:
+        # Drop events unrelated to value
+        events_dict = _drop(events_dict, 'n/a', 'value')
+        descriptions = np.asarray(events_dict['value'], dtype=str)
+
+    # Worst case, we go with 'n/a' for all events
+    else:
+        descriptions = 'n/a'
+
+    # Add Events to raw as annotations
+    onsets = np.asarray(events_dict['onset'], dtype=float)
+    durations = np.asarray(events_dict['duration'], dtype=float)
+    annot_from_events = mne.Annotations(onset=onsets,
+                                        duration=durations,
+                                        description=descriptions,
+                                        orig_time=raw.info['meas_date'])
+    raw.set_annotations(annot_from_events)
+
+    return raw
+
+
+def _handle_channels_reading(channels_fname, raw):
+    """Read associated channels.tsv and populate raw.
+
+    Updates status (bad) and types of channels.
+    """
+    channels_dict = _from_tsv(channels_fname)
+
+    # If we have a channels.tsv file, make sure there is the optional "status"
+    # column from which to infer good and bad channels
+    if 'status' in channels_dict:
+        # find bads from channels.tsv
+        bad_bool = [True if chn == 'bad' else False
+                    for chn in channels_dict['status']]
```

Comment:
This can lead to surprising results if there are typos in channels.tsv or one of them is in caps. Imagine if it's written as BAD instead of bad or if it's written Bad

Reply:
I agree, this should be solved with `chn.lower()`

Reply:
but ideally, these things should get caught by the BIDS-validator. I have opened several issues there that we need to scrutinize TSV files more closely:

- https://github.com/bids-standard/bids-validator/issues/787
- https://github.com/bids-standard/bids-validator/issues/777

I have also been in touch with Ross Blair from the Poldracklab, who is currently maintaining the validator, and we agreed that some funding will be directed to Squishymedia to work on these things :+1:

Reply:
sounds good, but doesn't hurt to have more checks. The scientific Python ecosystem is much more stronger than JS, so you can write more tight code.

Reply:
This would still mark anything that's not "bad" or "BAD" as "good". Is that fine?